extensions/git/src/commands.ts

```diff
@@ -2239,7 +2261,49 @@ export class CommandCenter {
 			return;
 		}
 
-		await repository.fetchDefault();
+		if (repository.remotes.length === 1) {
+			await repository.fetchDefault();
+			return;
+		}
+
+		const remoteItems: RemoteItem[] = repository.remotes.map(r => new RemoteItem(repository, r));
+
+		if (repository.HEAD?.upstream?.remote) {
+			// Move default remote to the top
+			const defaultRemoteIndex = remoteItems
+				.findIndex(r => r.remoteName === repository.HEAD!.upstream!.remote);
+
+			if (defaultRemoteIndex !== -1) {
+				remoteItems.splice(0, 0, ...remoteItems.splice(defaultRemoteIndex, 1));
+			}
+		}
+
+		const quickpick = window.createQuickPick();
+		quickpick.placeholder = localize('select a remote to fetch', 'Select a remote to fetch');
+		quickpick.canSelectMany = false;
+		quickpick.items = [...remoteItems, { label: '', kind: QuickPickItemKind.Separator }, new FetchAllRemotesItem(repository)];
+
+		quickpick.show();
+		const remoteItem = await new Promise<RemoteItem | FetchAllRemotesItem | undefined>(resolve => {
+			let isAccepted = false;
+
+			quickpick.onDidAccept(() => {
+				isAccepted = true;
+				resolve(quickpick.activeItems[0] as RemoteItem | FetchAllRemotesItem);
+			});
+			quickpick.onDidHide(() => {
+				if (!isAccepted) {
```

Comment:
Is this `isAccepted` variable really necessary? Calling `resolve` twice is a no-op.

Reply:
Makes sense. I have cleaned that up.